map.go

```diff
@@ -405,9 +405,9 @@ func (m *BPFMap) GetValueFlags(key unsafe.Pointer, flags MapFlag) ([]byte, error
 	return value, nil
 }
 
-// TODO: implement `bpf_map__lookup_and_delete_elem` wrapper
-// func (m *BPFMap) GetValueAndDeleteKey(key unsafe.Pointer) ([]byte, error) {
-// }
+func (m *BPFMap) GetValueAndDeleteKey(key unsafe.Pointer) ([]byte, error) {
+	return m.bpfMapLow.GetValueAndDeleteKey(key)
```

Comment:
As this has its own correlation in the libbpf API, we should use it instead of relying on the low level API.

`bpf_map__lookup_and_delete_elem`

Use, by instance, `(m *BPFMap) GetValueFlags` structure as a hint.

Reply:
> As this has its own correlation in the libbpf API, we should use it instead of relying on the low level API.
> 
> `bpf_map__lookup_and_delete_elem`
> 
> Use, by instance, `(m *BPFMap) GetValueFlags` structure as a hint.

done